src/test/java/com/iota/iri/network/impl/TransactionRequesterWorkerImplTest.java

```diff
@@ -0,0 +1,147 @@
+    
+    @Mock
+    private TipsViewModel tipsVM;
+
+    @Before
+    public void before() throws NoSuchFieldException, SecurityException, Exception {
+        tangle = new Tangle();
+        tangle.init();
+        
+        requester = new TransactionRequester(tangle, snapshotProvider, messageQ);
+        
+        worker = new TransactionRequesterWorkerImpl();
+        worker.init(tangle, requester, tipsVM, node);
+    }
+    
+    @After
+    public void tearDown() throws Exception {
+        worker.shutdown();
+    }
+    
+    @Test
+    public void workerActive() throws Exception {
+        assertFalse("Empty worker should not be active", worker.isActive());
+        
+        fillRequester();
+        
+        assertTrue("Worker should be active when it requester is over threshold", worker.isActive());
+    }
+    
+    @Test
+    public void processRequestQueueTest() throws Exception {
+        when(tipsVM.getRandomSolidTipHash()).thenReturn(
+                TVMRandomNull.getHash(), 
+                TVMRandomNotNull.getHash(), 
+                TVMAll9Null.getHash(), 
+                TVMAll9NotNull.getHash());
+        TangleMockUtils.mockTransaction(tangle, TVMRandomNull.getHash(), buildTransaction(TVMRandomNull.trits()));
+        TangleMockUtils.mockTransaction(tangle, TVMRandomNotNull.getHash(), buildTransaction(TVMRandomNotNull.trits()));
+        TangleMockUtils.mockTransaction(tangle, TVMAll9Null.getHash(), buildTransaction(TVMAll9Null.trits()));
+        TangleMockUtils.mockTransaction(tangle, TVMAll9NotNull.getHash(), buildTransaction(TVMAll9NotNull.trits()));
+        
+        addRequest(TVMRandomNull);
+        addRequest(TVMRandomNotNull);
+        addRequest(TVMAll9Null);
+        addRequest(TVMAll9NotNull);
+        
+        fillRequester();
+        
+        assertTrue("Null transaction hash should be processed", worker.processRequestQueue());
+        addRequest();
+        assertTrue("Not null transaction hash should be processed", worker.processRequestQueue());
+        addRequest();
+        assertTrue("Null transaction hash should be processed", worker.processRequestQueue());
+        addRequest();
+        assertFalse("All 9s transaction should not be processed", worker.processRequestQueue());
+    }
+
+    @Test
+    public void validTipToAddTest() throws Exception {
+       assertTrue("Null transaction hash should always be accepted", worker.isValidTransaction(TVMRandomNull));
+       assertTrue("Not null transaction hash should always be accepted", worker.isValidTransaction(TVMRandomNotNull));
+       assertTrue("Null transaction hash should always be accepted", worker.isValidTransaction(TVMAll9Null));
+       
+       assertFalse("All 9s transaction should not be accepted", worker.isValidTransaction(TVMAll9NotNull));
+    }
+    
+    private void fillRequester() throws Exception {
+        for (int i=0; i< TransactionRequesterWorkerImpl.REQUESTER_THREAD_ACTIVATION_THRESHOLD; i++) {
```

Comment:
I am trying to think...
Sometimes in unit tests we don't like to use the class constants becuase if a dev changes them, the test changes automatically. Sometimes we want to protect the code from devs casually changing constants.

However here it is ok, because my main worry is if someone decreases the constant, and hardcoding won't help.
So no need to change :-)